tee-worker/scripts/litentry/release/deploy.sh

```diff
@@ -0,0 +1,640 @@
+  request='{"jsonrpc":"2.0","id":1,"method":"chain_getHeader","params":[]}'
+
+  # Make the JSON-RPC request and retrieve the latest finalized block
+  response=$(curl -s -H "Content-Type: application/json" -d "$request" http://$PARACHAIN_HOST:$PARACHAIN_PORT)
+  hex_number=$(echo "$response" | grep -oP '(?<="number":")[^"]+')
+  dec_number=$(printf "%d" "$hex_number")
+
+
+  # Store the latest finalized block numbe  r in an environment variable
+  export LATEST_FINALIZED_BLOCK=${dec_number}
+
+  echo "Latest finalized block number: $LATEST_FINALIZED_BLOCK"
+}
+
+function build_parachain(){
+  cd $ROOTDIR
+  make build-node
+}
+
+function build_worker(){
+  cd $ROOTDIR/tee-worker/
+  source /opt/intel/sgxsdk/environment
+  SGX_COMMERCIAL_KEY=/home/faisal/litentry-parachain/tee-worker/enclave-runtime/Enclave_private.pem SGX_PRODUCTION=1 make
+}
+
+# Default values
+verbose=false
+build=false
+discard=false
+config=""
+auto_start=false
+export CHAIN=rococo
+export ONLY_WORKER=false
+export PARACHAIN_HOST="localhost"
+export PARACHAIN_PORT="9944"
+
+# Parse command-line options and arguments
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -h|--help)
+      display_help
+      exit 0
+      ;;
+    -b|--build)
+      build=true
+      shift
+      ;;
+    -d|--discard)
+      discard=true
+      shift
+      ;;
+    -c|--config)
+      if [[ $# -lt 2 ]]; then
+        echo "Error: The config file name is missing."
+        display_help
+        exit 1
+      fi
+      config="$2"
+      shift 2
+      ;;
+    -a|--only-worker)
+      export ONLY_WORKER=true
+      shift
+      ;;
+    -c| --chain)
+      export CHAIN="$2"
+      shift
+      ;;
+    -p| --parachain-port)
+      export PARACHAIN_PORT="$2"
+      shift
+      ;;
+    -h| --parachain-host)
+      export PARACHAIN_HOST="$2"
+      shift
+      ;;
+    restart|upgrade-worker)
+      action="$1"
+      shift
+      ;;
+    *)
+      echo "Error: Unknown option or argument '$1'."
+      display_help
+      exit 1
+      ;;
+  esac
+done
+
+# Create systemd folder for user if not already present
+mkdir -p ~/.config/systemd/user
+
+if [ -n "$config" ]; then
+  echo "Config file: $config"
+fi
+
+export CONFIG=$(cat $config)
+
+if [ "$discard" = true ]; then
+  echo "Cleaning the existing state for Parachain and Worker."
+  stop_running_services
+  rm -rf /tmp/parachain_dev/
+  rm -rf $ROOTDIR/tee-worker/tmp
+fi
+
+# Get Old MRENCLAVE
+if [ "$action" = "upgrade-worker" ]; then
+  cd $ROOTDIR/tee-worker
+  output=$(make mrenclave 2>&1)
+  if [[ $? -eq 0 ]]; then
+    mrenclave_value=$(echo "$output" | awk '{print $2}')
+    echo "MRENCLAVE value: $mrenclave_value"
+    export OLD_MRENCLAVE="$mrenclave_value"
+  else
+    echo "Failed to extract MRENCLAVE value."
+    exit 1
+  fi
+fi
+
+# Move log files  to log-backup
+if [ -d "log" ]; then
+  new_folder_name=$(date +"log-backup/log-%Y%m%d-%H%M%S")
+  cp -r "log" "$new_folder_name"
+  echo "Backup log into $new_folder_name"
+fi
```

Comment:
Can you backup everything or just the whole folder `tmp/w${i}` before doing any change?
cc @BillyWooo please confirm if that's enough

The backup should be carried out on both restart (even though it's a reset) and upgrade

Reply:
I took the logic to backup the logs from the python script for launching the worker, I used the same to back up the tmp folder 

```
if [ -d "$ROOTDIR/tee-worker/tmp/w$i" ]; then
          new_folder_name=$(date +"$ROOTDIR/tee-worker/tmp/w$i-%Y%m%d-%H%M%S")
          mkdir -p new_folder_name
          cp -r $ROOTDIR/tee-worker/tmp/w$i $new_folder_name
          echo "Backing up, previous worker binary $new_folder_name"
      fi
```
This will take place during restart and upgrade-worker